metricflow/model/validations/materializations.py

```diff
@@ -109,7 +109,18 @@ def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  #
         """Check that all of the metrics and dimensions listed in a materialization are valid."""
         issues: List[ValidationIssueType] = []
 
-        semantic_model = SemanticModel(model)
+        try:
+            semantic_model = SemanticModel(model)
+        except Exception as e:
```

Comment:
Yea my worry was it'll just turn in to a large swath of error types we end up trying to catch. I'll take a dive to see if there is an easy tight scoping. For instance it might be that handling `AssertionError` and `BaseBuildError` is enough assuming there is something like `BaseBuildError`

Reply:
I think we should be able to get away with `AssertionError` and `SemanticException`